ENH : Update the SOPs specifying the ET-to-BIDS conversion (#346)
TheAxonLab/hcph-sops: docs/data-management/post-session.md

    ```

### Convert eye-tracking into BIDS with *bidsphysio*
- [ ] Pull the latest docker image of esavary/bidsphysio with:

> ``` shell
> docker pull esavary/bidsphysio
> ```

Reviewer: ```suggestion
    ``` shell
    docker pull esavary/bidsphysio
    ```
```

> - [ ] Open [schedule_ET_data.tsv](../code/physioconv/schedule_ET_data.tsv) to find the phase encoding direction and the name of the .EDF file corresponding to the session you want to convert.

Reviewer: ```suggestion
- [ ] Open [`schedule_ET_data.tsv`](../code/physioconv/schedule_ET_data.tsv) to find the phase encoding direction and the name of the `.EDF` file corresponding to the session you want to convert.
```

> - [ ] In the folder containing your data, create a new folder named metadata. Copy the file containing information about the eye-tracker, named [info_ET.json](../code/physioconv/info_ET.json), inside this metadata folder.

Reviewer: Please start every sentence in a new line. Beyond the stylistic comment, I cannot understand this. Could you rephrase?
```suggestion
- [ ] In the folder containing your data, create a new folder named metadata.
    Copy the file containing information about the ET, named [`info_ET.json`](../code/physioconv/info_ET.json), inside this metadata folder.
```

> - [ ]  Run the following command for the file corresponding to the DWI (Diffusion-Weighted Imaging) data, replacing PATH_RAW_DATA,PATH_OUTPUT, DWI_FILENAME, SESSION_NUMBER, and PHASE_ENCODING_DIRECTION with the information corresponding to the file you want to process:

Reviewer: Please use real names (examples) rather than placeholders.
```suggestion
- [ ]  Run the following command for the file corresponding to the DWI (Diffusion-Weighted Imaging) data, replacing PATH_RAW_DATA,PATH_OUTPUT, DWI_FILENAME, SESSION_NUMBER, and PHASE_ENCODING_DIRECTION with the information corresponding to the file you want to process:
```

Author: I think it can make it more confusing as to what exactly needs to be changed and where when running the command, especially when directly including the session number and phase encoding direction. Would it not be better to include this in the notebook or as an example?

Reviewer: I don't think the notebook is a replacement for this. I'm happy also to have this example in the notebook, but I believe this is easier to understand with specific examples.

If it were only one field, then I'd probably be okay with it, but the number of parameters to replace is a mouthful and they only extend the example code (e.g., `PHASE_ENCODING_DIRECTION` instead of just `LR`).

Author: As you prefer. I've updated it with a real example.

Reviewer: Thx!

> ``` shell
> docker run -u $( id -u ):$( id -g ) --rm -it -v PATH_RAW_DATA:/data -v PATH_OUTPUT:/output --entrypoint=/opt/venv/bin/python esavary/bidsphysio /opt/venv/bin/edf2bidsphysio --infile DWI_FILENAME --bidsprefix /output/dwi/sub-001_ses-SESSION_NUMBER_acq-highres_dir-PHASE_ENCODING_DIRECTION -m /data/metadata/info_ET.json
> 
> ```

Reviewer: ```suggestion
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v PATH_RAW_DATA:/data -v PATH_OUTPUT:/output \
        --entrypoint=/opt/venv/bin/python esavary/bidsphysio \
        /opt/venv/bin/edf2bidsphysio --infile DWI_FILENAME \
        --bidsprefix /output/dwi/sub-001_ses-SESSION_NUMBER_acq-highres_dir-PHASE_ENCODING_DIRECTION \
        -m /data/metadata/info_ET.json
    ```
```

> !!! In addition to the eyetracker data and metadata JSON file, the code also generates a .tsv file containing all messages sent to the eye-tracker and the header of the .EDF file (named ..._eventlist_raw.tsv). Please exercise caution and ensure that you do not copy these generated files into the BIDS-compatible repository."

Reviewer: This will not render properly, which makes me think that should be checked before next iterations of review.
```suggestion
    !!! danger "Do not copy all output files directly"
    
        In addition to the eye-tracking data (`<prefix>_eyetrack.tsv.gz` file) and metadata (`<prefix>_eyetrack.json` file), the code also generates a TSV file containing all messages sent to the ET and the header of the `.EDF` file (with name `<prefix>_eventlist_raw.tsv`).
        Please do not copy these generated files into the *DataLad* dataset in BIDS.
```

> !!! warning "Instead of the current specifications, we will use [the following BEP](https://bids-specification--1128.org.readthedocs.build/en/1128/modality-specific-files/eye-tracking.html)"

Reviewer: This needs to be updated and possibly moved before the new edits.

    ```

### Convert eye-tracking into BIDS with *bidsphysio*
- [ ] Pull the latest docker image of esavary/bidsphysio with:
``` shell
docker pull esavary/bidsphysio
```
- [ ] Open [schedule_ET_data.tsv](../code/physioconv/schedule_ET_data.tsv) to find the phase encoding direction and the name of the .EDF file corresponding to the session you want to convert.
- [ ] In the folder containing your data, create a new folder named metadata. Copy the file containing information about the eye-tracker, named [info_ET.json](../code/physioconv/info_ET.json), inside this metadata folder.
- [ ]  Run the following command for the file corresponding to the DWI (Diffusion-Weighted Imaging) data, replacing PATH_RAW_DATA,PATH_OUTPUT, DWI_FILENAME, SESSION_NUMBER, and PHASE_ENCODING_DIRECTION with the information corresponding to the file you want to process:
``` shell
docker run -u $( id -u ):$( id -g ) --rm -it -v PATH_RAW_DATA:/data -v PATH_OUTPUT:/output --entrypoint=/opt/venv/bin/python esavary/bidsphysio /opt/venv/bin/edf2bidsphysio --infile DWI_FILENAME --bidsprefix /output/dwi/sub-001_ses-SESSION_NUMBER_acq-highres_dir-PHASE_ENCODING_DIRECTION -m /data/metadata/info_ET.json

```

> - [ ] Run the following command for the three files corresponding to the functional tasks:
> ``` shell
> docker run -u $( id -u ):$( id -g ) --rm -it -v PATH_RAW_DATA:/data -v PATH_OUTPUT:/output --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio --infile /data/FILENAME --bidsprefix /output/session01/func/sub-001_ses-SESSION_NUMBER_task-TASK_NAME_dir-PHASE_ENCODING_DIRECTION -m /data/metadata/info_ET.json
> ```

Reviewer: ```suggestion
- [ ] Run the following command for the three files corresponding to the functional tasks:
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v PATH_RAW_DATA:/data -v PATH_OUTPUT:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio \
        /opt/venv/bin/edf2bidsphysio \
        --infile /data/FILENAME \
        --bidsprefix /output/session01/func/sub-001_ses-SESSION_NUMBER_task-TASK_NAME_dir-PHASE_ENCODING_DIRECTION -m /data/metadata/info_ET.json
    ```
```

### Convert eye-tracking into BIDS with *bidsphysio*

!!! warning "Instead of the current specifications, we are using [the following BEP](https://bids-specification--1128.org.readthedocs.build/en/1128/modality-specific-files/eye-tracking.html)"

- [ ] Pull the latest docker image of esavary/bidsphysio with:
    ``` shell
    docker pull esavary/bidsphysio
    ```

> - [ ] Open [`schedule_ET_data.tsv`](../code/physioconv/schedule_ET_data.tsv) to find the phase encoding direction and the name of the `.EDF` file corresponding to the session you want to convert.

Reviewer: ```suggestion
- [ ] Open [`schedule.tsv`](../code/eyetracking/schedule.tsv) to find the phase encoding direction and the name of the `.EDF` file corresponding to the session you want to convert.
```

> - [ ] Create a new folder named metadata inside the folder containing your .EDF file.

Reviewer: Is this what you  mean?

```suggestion
- [ ] Create a new folder named `metadata/` inside the folder containing the target `.EDF` file.
```

>     Copy the file containing information about the ET, named [`info_ET.json`](../code/physioconv/info_ET.json), inside this metadata folder.

Reviewer: ```suggestion
    Copy the file containing information about the ET, named [`info_ET.json`](../code/eyetracking/info_ET.json) into the `metadata/` folder.
```

- [ ]  Run the following command for the file corresponding to the DWI (Diffusion-Weighted Imaging) data, replacing PATH_RAW_DATA,PATH_OUTPUT, DWI_FILENAME, SESSION_NUMBER, and PHASE_ENCODING_DIRECTION with the information corresponding to the file you want to process:
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
    -v /home/esavary/Projects/HCPH-BIDS_ET/data/:/data -v /home/esavary/Projects/HCPH-BIDS_ET/output/:/output \
    --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
    --infile /data/fixation_2023-10-20_18h48.03.561_5_session_1.EDF \
    --bidsprefix /output/session01/dwi/sub-001_ses-001_acq-highres_dir-LR -m /data/metadata/info_ET.json
    ```
- [ ] Run the following command for the files corresponding to the functional tasks:

>     ``` shell
>     docker run -u $( id -u ):$( id -g ) --rm -it \
>     -v /home/esavary/Projects/HCPH-BIDS_ET/data/:/data -v /home/esavary/Projects/HCPH-BIDS_ET/output/:/output \
>     --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
>     --infile /data/qct_2023-10-20_19h40.38.964_2_session_1.EDF \
>     --bidsprefix /output/session01/func/sub-001_ses-001_task-qct_dir-LR -m /data/metadata/info_ET.json
>     ```

Reviewer: ```suggestion
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v <path-to-EDF-data>:/data \
        -v <output-path-on-host>:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
        --infile /data/qct_2023-10-20_19h40.38.964_2_session_1.EDF \
        --bidsprefix /output/session01/func/sub-001_ses-001_task-qct_dir-LR \
        -m /data/metadata/info_ET.json
    ```
```

### Convert eye-tracking into BIDS with *bidsphysio*

!!! warning "Instead of the current specifications, we are using [the following BEP](https://bids-specification--1128.org.readthedocs.build/en/1128/modality-specific-files/eye-tracking.html)"

- [ ] Pull the latest docker image of esavary/bidsphysio with:
    ``` shell
    docker pull esavary/bidsphysio
    ```
- [ ] Open [`schedule_ET_data.tsv`](../code/physioconv/schedule_ET_data.tsv) to find the phase encoding direction and the name of the `.EDF` file corresponding to the session you want to convert.
- [ ] Create a new folder named metadata inside the folder containing your .EDF file.
    Copy the file containing information about the ET, named [`info_ET.json`](../code/physioconv/info_ET.json), inside this metadata folder.

> - [ ]  Run the following command for the file corresponding to the DWI (Diffusion-Weighted Imaging) data, replacing PATH_RAW_DATA,PATH_OUTPUT, DWI_FILENAME, SESSION_NUMBER, and PHASE_ENCODING_DIRECTION with the information corresponding to the file you want to process:

Reviewer: ```suggestion
- [ ]  Execute the ET to BIDS conversion on the dMRI data.
    Run the following command for the corresponding file in the [`schedule.tsv`](../code/eyetracking/schedule.tsv) file:
```

>     ``` shell
>     docker run -u $( id -u ):$( id -g ) --rm -it \
>     -v /home/esavary/Projects/HCPH-BIDS_ET/data/:/data -v /home/esavary/Projects/HCPH-BIDS_ET/output/:/output \
>     --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
>     --infile /data/fixation_2023-10-20_18h48.03.561_5_session_1.EDF \
>     --bidsprefix /output/session01/dwi/sub-001_ses-001_acq-highres_dir-LR -m /data/metadata/info_ET.json
>     ```

Reviewer: ```suggestion
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v <path-to-EDF-data>:/data \
        -v <output-path-on-host>:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
        --infile /data/fixation_2023-10-20_18h48.03.561_5_session_1.EDF \
        --bidsprefix /output/session01/dwi/sub-001_ses-001_acq-highres_dir-LR \
        -m /data/metadata/info_ET.json
    ```
```

- [ ] Run the following command for the files corresponding to the functional tasks:
    ``` shell
    
docker run -u $( id -u ):$( id -g ) --rm -it \
-v /home/esavary/Projects/HCPH-BIDS_ET/data/:/data -v /home/esavary/Projects/HCPH-BIDS_ET/output/:/output \
--entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
--infile /data/qct_2023-10-20_19h40.38.964_2_session_1.EDF \
--bidsprefix /output/session01/func/sub-001_ses-001_task-qct_dir-LR -m /data/metadata/info_ET.json

>     ```

Reviewer: Here, the spacings and indentations were removed. Whatever tool did this you should be careful not to use it or tame it:

```suggestion
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v <path-to-EDF-data>:/data \
        -v <output-path-on-host>:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
        --infile /data/qct_2023-10-20_19h40.38.964_2_session_1.EDF \
        --bidsprefix /output/session01/func/sub-001_ses-001_task-qct_dir-LR \
        -m /data/metadata/info_ET.json
    ```
```

Reviewer: This one was updated while I was writing. Marking resolved.

> !!! danger "Do not copy all output files directly"
>     In addition to the eye-tracking data (`<prefix>_eyetrack.tsv.gz` file) and metadata (`<prefix>_eyetrack.json` file), the code also generates a TSV file containing all messages sent to the ET and the header of the `.EDF` file (with name `<prefix>_eventlist_raw.tsv`).
>         Please do not copy these generated files into the *DataLad* dataset in BIDS.

Reviewer: For some reason, my suggestion has been modified. Without the correct blank lines and indentation, this will not render properly (please check before requesting another round):

```suggestion

!!! danger "Do not copy all output files directly"

    In addition to the eye-tracking data (`<prefix>_eyetrack.tsv.gz` file) and metadata (`<prefix>_eyetrack.json` file), the code also generates a TSV file containing all messages sent to the ET and the header of the `.EDF` file (with name `<prefix>_eventlist_raw.tsv`).
    Please do not copy these generated files into the *DataLad* dataset in BIDS.
```

Reviewer: BTW, this last admonition has the implicit message that *some* undisclosed files MUST be copied (as opposed those that don't), so I'm guessing one more checkbox indicating these files should be added somewhere around here.

        -v <path-to-EDF-data>:/data \
        -v <output-path-on-host>:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
        --infile /data/fixation_2023-10-20_18h48.03.561_5_session_1.EDF \
        --bidsprefix /output/session01/dwi/sub-001_ses-001_acq-highres_dir-LR \
        -m /data/metadata/info_ET.json
    ```
- [ ] Run the following command for the files corresponding to the functional tasks:
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v <path-to-EDF-data>:/data \
        -v <output-path-on-host>:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
        --infile /data/qct_2023-10-20_19h40.38.964_2_session_1.EDF \
        --bidsprefix /output/session01/func/sub-001_ses-001_task-qct_dir-LR \
        -m /data/metadata/info_ET.json
    ```

!!! danger "Do not copy all output files directly"

    In addition to the eye-tracking data (`<prefix>_eyetrack.tsv.gz` file) and metadata (`<prefix>_eyetrack.json` file), the code also generates a TSV file containing all messages sent to the ET and the header of the `.EDF` file (with name `<prefix>_eventlist_raw.tsv`).

>     Please do not copy these generated files into the *DataLad* dataset in BIDS.

Reviewer: BTW, this last admonition has the implicit message that some undisclosed files MUST be copied (as opposed those that don't), so I'm guessing one more checkbox indicating these files should be added somewhere around here.

- [ ]  Execute the ET to BIDS conversion on the dMRI data.
    Run the following command for the corresponding file in the [`schedule.tsv`](../code/eyetracking/schedule.tsv) file:
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v <path-to-EDF-data>:/data \
        -v <output-path-on-host>:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
        --infile /data/fixation_2023-10-20_18h48.03.561_5_session_1.EDF \
        --bidsprefix /output/session01/dwi/sub-001_ses-001_acq-highres_dir-LR \
        -m /data/metadata/info_ET.json
    ```
- [ ] Run the following command for the files corresponding to the functional tasks:
    ``` shell
    docker run -u $( id -u ):$( id -g ) --rm -it \
        -v <path-to-EDF-data>:/data \
        -v <output-path-on-host>:/output \
        --entrypoint=/opt/venv/bin/python bidsphysio /opt/venv/bin/edf2bidsphysio \
        --infile /data/qct_2023-10-20_19h40.38.964_2_session_1.EDF \
        --bidsprefix /output/session01/func/sub-001_ses-001_task-qct_dir-LR \
        -m /data/metadata/info_ET.json
    ```

> - [ ] Copy all _eyetrack.tsv.gz and .json generated into your copy of the *DataLad* dataset in BIDS.

Reviewer: Just a nit pick
```suggestion
- [ ] Copy all `<prefix>_eyetrack.tsv.gz` and `<prefix>_eyetrack.json` generated into your copy of the *DataLad* dataset in BIDS.
```
